crates/subspace-farmer/src/web_socket_rpc.rs

```diff
@@ -0,0 +1,82 @@
+use crate::rpc::{NewHead, RpcClient};
+use async_trait::async_trait;
+use jsonrpsee::types::traits::{Client, SubscriptionClient};
+use jsonrpsee::types::v2::params::JsonRpcParams;
+use jsonrpsee::types::{Error, Subscription};
+use jsonrpsee::ws_client::{WsClient, WsClientBuilder};
+use std::sync::Arc;
+use subspace_rpc_primitives::{
+    EncodedBlockWithObjectMapping, FarmerMetadata, SlotInfo, SolutionResponse,
+};
+
+/// `WsClient` wrapper.
+#[derive(Clone, Debug)]
+pub struct WebSocketRpc {
```

Comment:
Just an idea, take it or leave it, `web_socket_rpc.rs` => `ws_rpc.rs`.
```suggestion
pub(crate) struct WsRpc {
```

Reply:
Actually yes, but we need to add `client` to it since we will also have a server. And with that in mind, we might want to expand or structure it even further because it will look like the client for the server we'll provide :upside_down_face:

Reply:
I can rename it as you suggest. Would be better, I agree. `WebSocketRpc::WebSocketRpc;` is kind of long :D